src/views/Pages/Login/Login.js

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from 'react';
+import { Link, withRouter } from 'react-router-dom';
+// import { Offline, Online } from "react-detect-offline";
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { isAuthEmail, authenticateUser, isAuthUserType } from '../../../services/Auth';
+import {
+  Button, Card, CardBody, CardGroup, Col, Container, Form, Input, InputGroup,
+  InputGroupAddon, InputGroupText, Row
+} from 'reactstrap';
+import { fetchUserData } from '../../../store/actions/user';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email_signIn: '',
+      password_signIn: '',
+      keep_signIn: false,
+      username_signUp: '',
+      password_signUp: '',
+      password_repeat: '',
+      email_signUp: '',
+      loading: false
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState(
+      {
+        [e.target.id]: e.target.value
+      }
+    );
+  }
+
+
+  handleSubmit(e) {
+    e.preventDefault();
+    let { email_signIn, password_signIn } = this.state;
+    let data = JSON.stringify({ "username": email_signIn, "password": password_signIn });
+    console.log(data);
+    let { fetchData } = this.props;
+    fetchData(data, 'login');
+    // this.setState({
+    // 	email_signIn: '',
+    // 	password_signIn: '',
+    // 	keep_signIn: false,
+    // 	username_signUp: '',
+    // 	password_signUp: '',
+    // 	password_repeat: '',
+    // 	email_signUp: '',
+    // 	loading: true
+    // })
+  }
+
+  componentDidUpdate(prevProps) {
+    let { userData } = this.props.userData;
+    if (this.props !== prevProps) {
+      // authenticateUser('fashanutosin7@gmail.com', 'customer');
+      // console.log(this.props.userData)
+      if (isAuthEmail() && isAuthUserType()) {
+        this.props.history.push('/order')
+      } else {
+        if (userData.status === 'successful' && userData.errorcount === '0') {
+          authenticateUser(userData.email, 'customer')
+          this.props.history.push('/order')
+        };
+      };
+    };
+  }
+
+  render() {
+    let u = '';
+    let f = '';
+    let { email_signIn, password_signIn } = this.state;
+    let { loading, error, errorMessage, userData } = this.props.userData;
+    error === 'true' && errorMessage !== '' ? (f = errorMessage) :
+      (userData.errorcount !== '0' ? (f = userData.error) : (f = ''));
+    loading === 'done' ? (u = 'Login') :
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not nest ternary expressions.](https://app.codacy.com/gh/BuildForSDG/Team-064-Frontend/pullRequest?prid=5703258)